client/battle/BattleFieldController.cpp

```diff
@@ -47,6 +47,33 @@ BattleFieldController::BattleFieldController(BattleInterface & owner):
 	cellUnitMovementHighlight = IImage::createFromFile("UnitMovementHighlight.PNG", EImageBlitMode::COLORKEY);
 	cellUnitMaxMovementHighlight = IImage::createFromFile("UnitMaxMovementHighlight.PNG", EImageBlitMode::COLORKEY);
 
+	// load single edges
+	fullDamageRangeLimitImages[0b000001] = IImage::createFromFile("rangeHighlights/green/topLeft.PNG", EImageBlitMode::COLORKEY);

```

Comment:
Perhaps move into json animation config after all?

Reply:
I do not understand this one :(. You mean there is a specific json file where these images can be specified?

Reply:
What I mean is that instead of listing images in source code and loading them into a map you can use:
- CAnimation class in code
- json definition of "animation", like this one: https://github.com/vcmi/vcmi/blob/develop/Mods/vcmi/Sprites/PortraitsLarge.json

This will allow you to load all your images in like 2 lines of code without hardcoded list of paths.

Reply:
```c++
 // in header
std::unique_ptr<CAnimation> fullDamageRangeLimitImages;

// in c++ on init
fullDamageRangeLimitImages = std::make_unique<CAnimation>("battle/rangeHighlightsGreen");
fullDamageRangeLimitImages->preload();

// in c++ on render
std::shared_ptr<IImage> imageToShow = fullDamageRangeLimitImages->getImage(indexToShow);
```